test/test_wsgi.py

```diff
@@ -0,0 +1,47 @@
+import time
+from datetime import datetime
+
+import pytest
+from kw.platform import wsgi as uut
+from webob.request import BaseRequest
+
+
+def create_app(sleep_seconds=0):
+    def simple_app(environ, start_response):
+        if sleep_seconds:
+            time.sleep(sleep_seconds)
```

Comment:
I've tried and have no idea how to do that with freezegun. That's  because the middleware uses `time.time()` to measure request time and than sleeps for that time. Now, the problem is that if I use freezegun, `time.time()` always returns the same value, so by subtracting `now - later` I always get the wrong value `0` instead of a real number of seconds. How can I test the delay function if I cannot use `time.sleep` then?

Reply:
I think it monkeypatches time.sleep to move the fake clock without actually sleeping.

Reply:
Sadly, it doesn't, I've tested it, it waits for the given number of seconds. And also:

```
>  kw/platform/wsgi.py(28)_slowdown_request()
-> resp = req.get_response(app)  # sleep works here - it really stops for configured num. of seconds
(Pdb) time.time()
1562457600.0
(Pdb) n
> kw/platform/wsgi.py(29)_slowdown_request()
-> seconds = time.time() - before_time
(Pdb) time.time()
1562457600.0
(Pdb) n
> kw/platform/wsgi.py(30)_slowdown_request()
-> time.sleep(seconds)
(Pdb) seconds
0.0
```

The behavior is weird and I cannot figure out how it really works.

Reply:
I used freezegun in the end, it has the `tick` argument which works nicely, but it still waits during sleep.